src/webgpu/api/validation/encoding/createRenderBundleEncoder.spec.ts

```diff
@@ -0,0 +1,161 @@
+export const description = `
+createRenderBundleEncoder validation tests.
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import {
+  kAllTextureFormats,
+  kDepthStencilFormats,
+  kTextureFormatInfo,
+  kMaxColorAttachments,
+} from '../../../capability_info.js';
+import { ValidationTest } from '../validation_test.js';
+
+export const g = makeTestGroup(ValidationTest);
+
+g.test('attachment_state')
+  .desc(
+    `
+    Tests that createRenderBundleEncoder correctly validates the attachment state passed to it.
+      - Must be <= kMaxColorAttachments (8) colorFormats
+      - Must have a depthStencilFormat if no colorFormats are given
+    `
+  )
+  .params(u =>
+    u //
+      .combine('colorFormatCount', [0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
```

Comment:
could do

```suggestion
      .combine('colorFormatCount', (new Array(kMaxColorAttachments + 2)).fill().map((_, i) => i))
```

maybe that's too magic though

Reply:
How about slightly shorter magic:
```js
[...Array(kMaxColorAttachments+2).keys()]
```